src/test/java/mate/academy/jvbookstore/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,237 @@
+                result.getResponse().getContentAsString(), CartItemDto.class);
+
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(cartItemId, actual.getId());
+        Assertions.assertEquals(cartItems.get(0).getBook().getId(), actual.getBookId());
+        Assertions.assertEquals(cartItems.get(0).getBook().getTitle(), actual.getBookTitle());
+        Assertions.assertEquals(requestDto.getQuantity(), actual.getQuantity());
+    }
+
+    @Test
+    @WithUserDetails(value = "owner@bookstore.com")
+    void deleteBookFromShoppingCart_WithCorrectId_Success(
+            @Autowired ShoppingCartRepository shoppingCartRepository,
+            @Autowired CartItemRepository cartItemRepository) throws Exception {
+        ownersShoppingCart.getCartItems().add(cartItems.get(0));
+        shoppingCartRepository.save(ownersShoppingCart);
+
+        Long cartItemId = cartItemRepository.findAll().get(0).getId();
+
+        mockMvc.perform(delete("/cart/cart-items/" + cartItemId))
+                .andExpect(status().isNoContent())
+                .andReturn();
+
+        boolean actual = cartItemRepository.findAll().isEmpty();
+
+        Assertions.assertTrue(actual);
+    }
+
+    @Test
+    @WithUserDetails(value = "owner@bookstore.com")
+    void clearShoppingCart_NoParams_Success(
+            @Autowired ShoppingCartRepository shoppingCartRepository,
+            @Autowired CartItemRepository cartItemRepository) throws Exception {
+
+        ownersShoppingCart.getCartItems().add(cartItems.get(0));
+        ownersShoppingCart.getCartItems().add(cartItems.get(1));
+        shoppingCartRepository.save(ownersShoppingCart);
+
+        mockMvc.perform(delete("/cart/cart-items"))
+                .andExpect(status().isNoContent())
+                .andReturn();
+
+        boolean actual = cartItemRepository.findAll().isEmpty();
+
+        Assertions.assertTrue(actual);
+    }
+
+    @AfterEach
+    void clearShoppingCart(@Autowired DataSource dataSource) throws SQLException {
+        ownersShoppingCart.setCartItems(new HashSet<>());
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("db/shoppingcart/clear-shopping-cart.sql"));
+        }
+    }
+
+    @AfterAll
+    static void clearDb(@Autowired DataSource dataSource) throws SQLException {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("db/book/delete-every-book.sql"));
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("db/shoppingcart/clear-shopping-cart.sql"));
+        }
+    }
```

Comment:
if you want to test making changes in DB, it's not required to execute SQL scripts for restoring DB after tests, you can use `@Transactional @Rollback` on method annotations or test containers